### DISCUSSION TITLE
use `.append` to add to album?

### DISCUSSION THREAD

**Discussion by frankrolf** [RhetTbull/osxphotos#1523] [Ideas]
It took me quite a while to find out how to create a new album, and how to add specific photos to it (using the `repl`).

For posterity, this is what worked:
```
my_photos = photosdb.photos(albums=['GR3'])
edited_photos = [p for p in my_photos if p.adjustments]
album = osxphotos.PhotosAlbum('GR3 edited')
album.add_list(edited_photos)
```

I tried a lot of other things, for example this (which does not work):
```
for p in edited_photos:
    p.albums.append('GR3 edited')
```

The most natural approach (for me) would be something like this:
```
album = osxphotos.PhotosAlbum('GR3 edited')
for p in edited_photos:
    album.append(p)
```

Does this sound useful?

An example of how to add things to albums would be much appreciated (after going through a lot of documentation, I found my solution here: https://github.com/RhetTbull/osxphotos/discussions/1487#discussioncomment-8991852)

**Comment by RhetTbull** [comment]
This is probably worth adding to the [API Docs](https://github.com/RhetTbull/osxphotos/blob/main/API_README.md) as it's a useful feature. It's not currently because in my mind, this isn't a core osxphotos feature. With the exception of `timewarp` and `batch-edit` commands (which aren't exposed in the public API), osxphotos provides *read only* access to the Photos library. It is not a goal of mine to enable the API to modify the library.  I've built some tools like `PhotosAlbum` as internal helpers to do modification where necessary but these are actually implemented using another library of mine, [PhotoScript](https://github.com/RhetTbull/PhotoScript). 

```
for p in edited_photos:
    p.albums.append('GR3 edited')
```

This syntax won't be supported in osxphotos because it would require a complete rewrite of the API which is predicated on the `PhotoInfo` object being read only. 

It does however make sense to add `append` and `extend` to the PhotosAlbum code and to expose this as a public API of osxphotos. So few people use the API that most of my effort on the CLI which is widely used.

**Comment by frankrolf** [comment]
Thank you – this sounds great! The “did not work” example (which would require an API rewrite) isn’t necessarily a request – it just was something I figured might work.

Adding `append` and `extend` to PhotosAlbum would be amazing – thank you in advance!
